src/sharing/ShareWrapper.js

```diff
@@ -170,6 +172,36 @@ class ShareWrapper extends React.Component<Props, State> {
     }
   };
 
+  deleteItem = toDelete => {
+    const files = [...this.state.files];
+    const filteredItems = files.filter(item => item.url !== toDelete.url);
+    this.setState({ files: filteredItems });
```

Comment:
This should get merged into a `setState` call with a callback:
https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous